crates/bevy_pbr/src/render/pbr_prepass.wgsl

```diff
@@ -1,87 +1,30 @@
-#import bevy_pbr::prepass_bindings
+#import bevy_pbr::pbr_prepass_functions
 #import bevy_pbr::pbr_bindings
 #import bevy_pbr::pbr_types
 #ifdef NORMAL_PREPASS
 #import bevy_pbr::pbr_functions
 #endif // NORMAL_PREPASS
 
-struct FragmentInput {
-    @builtin(front_facing) is_front: bool,
-    @builtin(position) frag_coord: vec4<f32>,
-#ifdef VERTEX_UVS
-    @location(0) uv: vec2<f32>,
-#endif // VERTEX_UVS
-
-#ifdef NORMAL_PREPASS
-    @location(1) world_normal: vec3<f32>,
-#ifdef VERTEX_TANGENTS
-    @location(2) world_tangent: vec4<f32>,
-#endif // VERTEX_TANGENTS
-#endif // NORMAL_PREPASS
-
-#ifdef MOTION_VECTOR_PREPASS
-    @location(3) world_position: vec4<f32>,
-    @location(4) previous_world_position: vec4<f32>,
-#endif // MOTION_VECTOR_PREPASS
-
-#ifdef DEPTH_CLAMP_ORTHO
-    @location(5) clip_position_unclamped: vec4<f32>,
-#endif // DEPTH_CLAMP_ORTHO
-};
-
-// Cutoff used for the premultiplied alpha modes BLEND and ADD.
-const PREMULTIPLIED_ALPHA_CUTOFF = 0.05;
-
-// We can use a simplified version of alpha_discard() here since we only need to handle the alpha_cutoff
-fn prepass_alpha_discard(in: FragmentInput) {
-
-#ifdef MAY_DISCARD
-    var output_color: vec4<f32> = bevy_pbr::pbr_bindings::material.base_color;
-
-#ifdef VERTEX_UVS
-    if (bevy_pbr::pbr_bindings::material.flags & bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_BASE_COLOR_TEXTURE_BIT) != 0u {
-        output_color = output_color * textureSampleBias(bevy_pbr::pbr_bindings::base_color_texture, bevy_pbr::pbr_bindings::base_color_sampler, in.uv, bevy_pbr::prepass_bindings::view.mip_bias);
-    }
-#endif // VERTEX_UVS
-
-    let alpha_mode = bevy_pbr::pbr_bindings::material.flags & bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_ALPHA_MODE_RESERVED_BITS;
-    if alpha_mode == bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_ALPHA_MODE_MASK {
-        if output_color.a < bevy_pbr::pbr_bindings::material.alpha_cutoff {
-            discard;
-        }
-    } else if (alpha_mode == bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BLEND || alpha_mode == bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_ALPHA_MODE_ADD) {
-        if output_color.a < PREMULTIPLIED_ALPHA_CUTOFF {
-            discard;
-        }
-    } else if alpha_mode == bevy_pbr::pbr_types::STANDARD_MATERIAL_FLAGS_ALPHA_MODE_PREMULTIPLIED {
-        if all(output_color < vec4(PREMULTIPLIED_ALPHA_CUTOFF)) {
-            discard;
-        }
-    }
-
-#endif // MAY_DISCARD
-}
-
+#import bevy_pbr::prepass_io as prepass_io
+
+// TODO Griffin using mesh_view_bindings here because of:
```

Comment:
This is a TODO but is it fixable? Maybe @robtfm's input is useful?

Reply:
Woops this TODO is old. I got rid of view in the prepass bindings altogether so we have just one source.